File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res = {}

        for i in range(len(nums)):
            diff = target - nums[i]

            if diff in res:
                return [res[diff], i]
            else:
                res[nums[i]] = i
        

File: Data Structures & Algorithms/guess-number-higher-or-lower/submission-2.js
/**
 * Forward declaration of guess API.
 * @param {number} num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * function guess(num) {}
 */

class Solution {
    /**
     * @param {number} n
     * @return {number}
     */
    guessNumber(n) {
        let l = 1;
        let r = n;

        while (l <= r) {
            const m = Math.floor((r + l) / 2);
            const res = guess(m);

            // m is lower than target
            if (res === 1) {
                l = m + 1;
            } else if (res === 0) {
                return m;
            } else {
                r = m - 1;
            }
        }

    }
}
